tests/testthat/test-class-spec.R

```diff
@@ -81,34 +130,38 @@ test_that("can work with simple S3 classes", {
   klass <- s3_data.frame
   expect_equal(as_class(klass), klass)
 
-  expect_equal(class_type(klass), "s3")
-  expect_equal(class_names(klass), c("R7_object", "data.frame"))
+  expect_equal(class_type(klass), "r7_s3")
+  expect_equal(class_dispatch(klass), c("R7_object", "data.frame"))
+  expect_equal(class_register(klass), "data.frame")
   expect_equal(class_desc(klass), "S3<data.frame>")
   expect_equal(class_construct(klass, list(x = 1)), data.frame(x = 1))
   expect_equal(class_deparse(klass), 's3_class("data.frame")')
 
   obj <- data.frame()
   expect_equal(obj_type(obj), "s3")
   expect_equal(obj_desc(obj), "S3<data.frame>")
+  expect_equal(obj_dispatch(obj), "data.frame")
   expect_equal(class_inherits(obj, klass), TRUE)
 })
 
-test_that("can work with compound s3 classes", {
+test_that("can work with s3 subclasses", {
   klass <- s3_class(
     class = c("ordered", "factor"),
     constructor = function(.data = numeric(), levels) ordered(.data, levels)
   )
   expect_equal(as_class(klass), klass)
 
-  expect_equal(class_type(klass), "s3")
-  expect_equal(class_names(klass), c("R7_object", "ordered", "factor"))
+  expect_equal(class_type(klass), "r7_s3")
+  expect_equal(class_dispatch(klass), c("R7_object", "ordered", "factor"))
+  expect_equal(class_register(klass), "ordered")
```

Comment:
Oooooh interesting I think I answered my own question from above.

And if you made an R7 subclass of ordered, then this looks good too

``` r
klass <- s3_class(
  class = c("ordered", "factor"),
  constructor = function(.data = numeric(), levels) ordered(.data, levels)
)

subklass <- new_class("ordered2", parent = klass)

class_dispatch(subklass)
#> [1] "ordered2"  "R7_object" "ordered"   "factor"
class_register(subklass)
#> [1] "ordered2"

obj <- subklass(1L, "x")
obj_dispatch(obj)
#> [1] "ordered2"  "R7_object" "ordered"   "factor"
```